src/shell/rlimit.c

```diff
@@ -0,0 +1,122 @@
+ *
+ * Call setrlimit(2) for any resource limits defined in
+ * attributes.system.shell.options.rlimit
+ */
+#define FLUX_SHELL_PLUGIN_NAME "rlimit"
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <signal.h>
+#include <sys/resource.h>
+#include <jansson.h>
+
+#include <flux/core.h>
+#include <flux/shell.h>
+
+#include "ccan/str/str.h"
+
+#include "internal.h"
+#include "builtins.h"
+
+static int rlimit_name_to_string (const char *name)
+{
+    if (streq (name, "RLIMIT_CPU"))
+        return RLIMIT_CPU;
+    if (streq (name, "RLIMIT_FSIZE"))
+        return RLIMIT_FSIZE;
+    if (streq (name, "RLIMIT_DATA"))
+        return RLIMIT_DATA;
+    if (streq (name, "RLIMIT_STACK"))
+        return RLIMIT_STACK;
+    if (streq (name, "RLIMIT_CORE"))
+        return RLIMIT_CORE;
+    if (streq (name, "RLIMIT_NOFILE"))
+        return RLIMIT_NOFILE;
+    if (streq (name, "RLIMIT_OFILE"))
+        return RLIMIT_OFILE;
+    if (streq (name, "RLIMIT_AS"))
+        return RLIMIT_AS;
+    if (streq (name, "RLIMIT_RSS"))
+        return RLIMIT_RSS;
+    if (streq (name, "RLIMIT_NPROC"))
+        return RLIMIT_NPROC;
+    if (streq (name, "RLIMIT_MEMLOCK"))
+        return RLIMIT_MEMLOCK;
+    if (streq (name, "RLIMIT_MSGQUEUE"))
+        return RLIMIT_MSGQUEUE;
+    if (streq (name, "RLIMIT_NICE"))
+        return RLIMIT_NICE;
+    if (streq (name, "RLIMIT_RTPRIO"))
+        return RLIMIT_RTPRIO;
+    if (streq (name, "RLIMIT_RTTIME"))
+        return RLIMIT_RTTIME;
+    if (streq (name, "RLIMIT_SIGPENDING"))
+        return RLIMIT_SIGPENDING;
+    return -1;
+}
+
+static int rlimit_init (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int rc = 0;
+    json_t *value;
+    const char *key;
+    json_t *o = NULL;
+    flux_shell_t *shell = flux_plugin_get_shell (p);
+
+    if (!shell)
+        return -1;
+
+    switch (flux_shell_getopt_unpack (shell, "rlimit", "o", &o)) {
+        case -1:
+            return shell_log_errno ("failed to parse rlimit shell option");
+        case 0:
+            return 0;
+        case 1:
+            break;
+    }
+    json_object_foreach (o, key, value) {
+        int resource;
+        struct rlimit rlim;
+        if (!json_is_integer (value)
+            || (resource = rlimit_name_to_string (key)) < 0) {
+            char *s = json_dumps (value, JSON_ENCODE_ANY);
+            shell_log_error ("invalid shell option rlimit.%s%s%s",
+                             key,
+                             s ? "=" : "",
+                             s ? s : "");
+            free (s);
+            rc = -1;
+            continue;
+        }
+        if (getrlimit (resource, &rlim) < 0)
+            return shell_log_errno ("getrlimit %s", key);
+        rlim.rlim_cur = json_integer_value (value);
+        if (setrlimit (resource, &rlim) < 0)
+            return shell_log_errno ("setrlimit %s", key);
```

Comment:
I guess logging this error and continuing makes sense, b/c a soft value that works on a login node may not be permitted on a compute node.

Reply:
Good point! I'll fix that.